actionview/lib/action_view/helpers/form_tag_helper.rb

```diff
@@ -829,6 +829,39 @@ def utf8_enforcer_tag
         '<input name="utf8" type="hidden" value="&#x2713;" />'.html_safe
       end
 
+      # Creates a text input linked to a datalist. A datalist contains a set of option tags.
+      # The datalist's behavior is similar to a select tag. Datalists allow a selection from a list of predefined values
+      # but are not constrained to those values. The text input can be chosen from the datalist options or entered normally.
+      #
+      # === Examples
+      #   datalist_field_tag 'browser'
+      #   # => <input type="text" name="browser" id="browser" list="browsers"><datalist id="browsers"></datalist>
+      #
+      #   datalist_field_tag 'browser', 'default browser'
+      #   # => <input type="text" name="browser" id="browser" value="default browser" list="browsers"><datalist id="browsers"></datalist>
+      #
+      #   datalist_field_tag 'browser', nil, '<option value="Chrome"></option><option value="Firefox"></option>'.html_safe
+      #   # => <input type="text" name="browser" id="browser" list="browsers"><datalist id="browsers"><option value="Chrome"></option><option value="Firefox"></option></datalist>
+      #
+      #   datalist_field_tag 'browser', nil, '<option>Chrome</option><option>Firefox</option>'.html_safe
+      #   # => <input type="text" name="browser" id="browser" list="browsers"><datalist id="browsers"><option>Chrome</option><option>Firefox</option></datalist>
+      #
+      #   datalist_field_tag 'browser', nil, options_from_collection_for_select(@browsers, 'id', 'name')
+      #   # => <input type="text" name="browser" id="browser" list="browsers"><datalist id="browsers"><option value="1">Chrome</option><option value="2">Firefox</option></datalist>
+      #
+      #   datalist_field_tag 'browser', nil, options_from_collection_for_select(@browsers, 'id', 'name'), placeholder: "placeholder", class: "browser-input"
+      #   # => <input type="text" name="browser" id="browser" placeholder="placeholder" class="browser-input" list="browsers"><datalist id="browsers"><option value="1">Chrome</option><option value="2">Firefox</option></datalist>
+      def datalist_field_tag(name, value = nil, option_tags = nil, options = {})
+        option_tags ||= ""
+        list_name = sanitize_to_id(name.to_s).pluralize
+
+        # Delete :list from text_field_options if they tried to include them
+        # Text field's list attribute should correspond to the datalist's id
+        options.delete(:list)
```

Comment:
Why would people be likely to include `:list` themselves?

We don't have to be overprotective: people can see list is overridden in the output HTML.